### ISSUE TITLE
Support Function data type in the SBUnit-Extensions generally

### ISSUE THREAD

**Issue by sbarber2** [streambase/SBUnit-Extensions#3]
There's another Issue to make sure JSONMatcherBuilder supports Functions, but the rest of the project needs to be reviewed and the Function data type handled where that makes sense.

Some places that come to mind:
CSVMatcherBuilder -- at least needs a test, might need code support
FieldBaseTupleMatcherTest -- needs function field tests

**Comment by haydens** [comment]
Steve,

I'm really happy to see you looking into these issues. Thanks.

What support for user functions do we need? Anything more than enqueueing a user function on a field? 

   Thanks,
      Hayden

**Comment by sbarber2** [comment]
Probably need all the relevant Matchers and MatcherBuilders to work on function field values, too, if they don't. At the least we need tests.

Note that this ticket is for the Unit Test Extensions, not SBUnit itself, so it doesn't cover enqueue or the Dequeuers or the Expecters. Those may already work with functions IDK.

**Comment by haydens** [comment]
But what does the mean, exactly? Test to see that the user functions are identical? I hope it’s not equivalent. Does == work with user functions in SB? If so what does it do?

I’m not really sure why I’d want to write a test that verifies that the user function dequeued is the same. I can think of some examples, but they’re fairly contrived. 

I’ve never written any code that dequeues a user function, so maybe I’m missing a trick.

**Comment by sbarber2** [comment]
Well, I'm not really thinking about WHY I might ever want to dequeue a function value. I just know that it is possible and we ought to verify that the Matchers etc. won't blow up if there were ever a test that wanted to.

That said, it's not obvious at first glance how to create an object that implements com.streambase.sb.Function . . . .

Oh, well, no one said this was a high priority issue . . . .

**Comment by sbarber2** [comment]
Indeed. Functions work fine with SBJUnit, the SB Java API, and even the Studio test generator wizard already:

``` java
    @Test
    public void testDequeueFunction() throws Exception {
        server.getEnqueuer("In")
                .enqueue(JSONSingleQuotesTupleMaker.MAKER, "{}");
        new Expecter(server.getDequeuer("Out"))
                .expect(JSONSingleQuotesTupleMaker.MAKER,
                        "{'f':{'function_definition':'function (x int)-> int { x * x }'}}");
    }
```
